src/OAuth2Form.tsx

```diff
@@ -302,9 +257,31 @@ export const OAuth2Form = () => {
 
   //init
   useEffect(() => {
-    getUserProfile();
+    const getOauthClientInfo = () => {
+      if (!oauthClientInfo && !error) {
+        const code = getURLParam("code");
+        if (code) return; // we're in the middle of a SSO, do not attempt to get OAuthClient info yet
+
+        const clientId = getURLParam("client_id");
+        if (!clientId) {
+          onError(new Error("Synapse OAuth client_id is required"));
+          return;
+        }
+        SynapseClient.getOAuth2Client(clientId)
+          .then((oauthClientInfo: OAuthClientPublic) => {
+            if (oauthClientInfo.verified) {
+              getOAuth2RequestDescription();
+            }
+            setOauthClientInfo(oauthClientInfo);
+          })
+          .catch((_err) => {
+            onError(_err);
+          });
+      }
+    };
+
     getOauthClientInfo();
-  }, [accessToken]);
+  }, []);
```

Comment:
Not fetching the userProfile here anymore. Just fetching the OAuth client, so it doesn't depend on the accessToken.